tasks/eva_2850/fix_discordant_variants.py

```diff
@@ -0,0 +1,337 @@
+                                     f"\n {rs_without_map_weight}")
+                        continue
+
+                    ss_records = all_ss_variants[rs]
+                    if not ss_records:
+                        logger.error(f"No original SS record found for RS {rs}, /n {ss_records}")
+                        continue
+
+                    rs_variant = rs_records[0]
+
+                    if check_all_ss_has_same_info(ss_records):
+                        if rs_variant['start'] == ss_records[0]['start']:
+                            logger.error(f"RS's and original SS's Start matches. Nothing to do")
+                            continue
+
+                        process_all_ss_has_same_info(rs_variant, ss_records)
+
+                    else:
+                        logger.error(
+                            f"For RS {rs}, Not all original SS has same info. Case for Split: \nSS Records {ss_records}")
+
+                rs_list = rs_list_to_process.copy()
+                rs_list_to_process.clear()
+
+
+
+
+def process_all_ss_has_same_info(rs_variant, ss_records):
+    rs_with_new_start = copy.copy(rs_variant)
+    rs_with_new_start['start'] = ss_records[0]['start']
+    rs_with_new_start['_id'] = get_clustered_SHA1(rs_with_new_start)
+
+    variant_in_db = check_for_hash_collision(rs_with_new_start['_id'])
+
+    if variant_in_db:
+        logger.warn(f"Hash collision will occur for RS {rs_variant['accession']} with RS {variant_in_db['accession']}")
+        resolve_collision_and_insert_rs(rs_variant, rs_with_new_start, variant_in_db)
+    else:
+        logger.info(f"No hash collision for RS {rs_variant['accession']}")
+        correct_rs_start_and_insert(rs_variant, rs_with_new_start)
+
+
+def correct_rs_start_and_insert(rs_variant, rs_with_new_start):
+    dbsnp_cve_collection = mongo_source.mongo_handle[mongo_source.db_name][DBSNP_CLUSTERED_VARIANT_ENTITY]
+
+    drop_statements = []
+    insert_statements = []
+    drop_statements.append(pymongo.DeleteOne({'_id': rs_variant['_id']}))
+    insert_statements.append(pymongo.InsertOne(rs_with_new_start))
+
+    result_drop = dbsnp_cve_collection.bulk_write(requests=drop_statements, ordered=False)
+    logger.info('There was %s old documents dropped' % result_drop.deleted_count)
+    result_insert = dbsnp_cve_collection.bulk_write(requests=insert_statements, ordered=False)
+    logger.info('There was %s new documents inserted' % result_insert.inserted_count)
+
+
+def resolve_collision_and_insert_rs(rs_variant, rs_with_new_start, variant_in_db):
+    dbsnp_cve_collection = mongo_source.mongo_handle[mongo_source.db_name][DBSNP_CLUSTERED_VARIANT_ENTITY]
+    dbsnp_cvoe_collection = mongo_source.mongo_handle[mongo_source.db_name][DBSNP_CLUSTERED_VARIANT_OPERATION_ENTITY]
+
+    dbsnp_cve_drop_statements = []
+    dbsnp_cve_insert_statements = []
+    dbsnp_cvoe_insert_statements = []
+
+    if rs_with_new_start['accession'] < variant_in_db['accession']:
```

Comment:
Suggest linking to the [policy](https://github.com/EBIvariation/eva-accession/blob/0b2ae4cdb6f74152c5443c3831c02c1d76cf93f9/eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteredVariantMergingPolicy.java#L40)

Reply:
done